Update Chronograf 1.8 Flux template variables (#2777)
influxdata/docs-v2: content/chronograf/v1.8/guides/dashboard-template-variables.md

After you set up variables, variables are available to select in your dashboard user interface (UI).

- [Use template variables in cell queries](#use-template-variables-in-cell-queries)
  - [InfluxQL](#influxql)
  - [Flux](#flux)
- [Use template variables in cell titles](#use-template-variables-in-cell-titles)

![Use template variables](/img/chronograf/1-6-template-vars-use.gif)

### Use template variables in cell queries
Both InfluxQL and Flux support template variables.

#### InfluxQL
In an InfluxQL query, surround template variables names with colons (`:`) as follows:

```sql
SELECT :variable_name: FROM "telegraf"."autogen".:measurement: WHERE time < :dashboardTime:
```

##### Quoting template variables in InfluxQL

For **predefined meta queries** such as "Field Keys" and "Tag Values", **do not add quotes** (single or double) to your queries. Chronograf will add quotes as follows:

```sql
SELECT :variable_name: FROM "telegraf"."autogen".:measurement: WHERE time < :dashboardTime:
```

> For **custom queries**, **CSV**, or **map queries**, quote the values in the query following standard [InfluxQL](/{{< latest "influxdb" "v1" >}}/query_language/) syntax as follows:

Author: ```suggestion
For **custom queries**, **CSV**, or **map queries**, quote the values in the query following standard [InfluxQL](/{{< latest "influxdb" "v1" >}}/query_language/) syntax:
```

WHERE time > :dashboardTime:
```

{{% note %}}

> To use the date picker to specify a particular time range in the past
> which does not include "now", construct the query using `:dashboardTime:`
> as the lower limit and [`:upperDashboardTime:`](#upperdashboardtime) as the upper limit.

Author: ```suggestion
To use the date picker to specify a particular time range in the past
which does not include "now", construct the query using `:dashboardTime:`
as the start time and [`:upperDashboardTime:`](#upperdashboardtime) as the stop time.
```

WHERE time > :dashboardTime:
```

{{% note %}}
In order to use the date picker to specify a particular time range in the past

> which does not include "now", the query should be constructed using `:dashboardTime:`

Reviewer: ```suggestion
which does not include "now", construct the query using `:dashboardTime:`
```

WHERE time > :dashboardtime:
GROUP BY time(:interval:)
```
{{% /tab-content %}}
{{% tab-content %}}
- [`dashboardTime`](#dashboardtime-flux)
- [`upperDashboardTime`](#upperdashboardtime-flux)
- [`autoInterval`](#autointerval)

### dashboardTime {id="dashboardtime-flux"}

The `dashboardTime` template variable is controlled by the "time" dropdown in your Chronograf dashboard.

<img src="/img/chronograf/1-6-template-vars-time-dropdown.png" style="width:100%;max-width:549px;" alt="Dashboard time selector"/>

If using relative time, this variable represents the time offset specified in the dropdown (-5m, -15m, -30m, etc.) and assumes time is relative to "now".

> If using absolute time defined by the date picker, `dashboardTime` is populated with lower threshold.

Author: ```suggestion
If using absolute time defined by the date picker, `dashboardTime` is populated with the start time.
```

```js
from(bucket: "telegraf/autogen")
  |> range(start: dashboardTime)
  |> filter(fn: (r) => r._measurement == "cpu" and r._field == "usage_system")
```

{{% note %}}
To use the date picker to specify a time range in the past without "now", use `dashboardTime` as the lower limit and
[`upperDashboardTime`](#upperdashboardtime-flux) as the upper limit.
{{% /note %}}

### upperDashboardTime{id="upperdashboardtime-flux"}

> The `upperDashboardTime` template variable is defined by the upper time limit specified using the date picker.

Author: ```suggestion
The `upperDashboardTime` template variable is defined by the stop time specified using the date picker.
```

<img src="/img/chronograf/1-6-template-vars-date-picker.png" style="width:100%;max-width:762px;" alt="Dashboard date picker"/>

> For relative time frames, this variable inherits `now()`. For absolute time frames, this variable inherits the upper time limit.

Author: ```suggestion
For relative time frames, this variable inherits `now()`. For absolute time frames, this variable inherits the stop time.
```

WHERE time > :dashboardtime:
GROUP BY time(:interval:)
```
{{% /tab-content %}}
{{% tab-content %}}
- [`dashboardTime`](#dashboardtime-flux)
- [`upperDashboardTime`](#upperdashboardtime-flux)
- [`autoInterval`](#autointerval)

### dashboardTime {id="dashboardtime-flux"}

The `dashboardTime` template variable is controlled by the "time" dropdown in your Chronograf dashboard.

<img src="/img/chronograf/1-6-template-vars-time-dropdown.png" style="width:100%;max-width:549px;" alt="Dashboard time selector"/>

If using relative time, this variable represents the time offset specified in the dropdown (-5m, -15m, -30m, etc.) and assumes time is relative to "now".

> If using absolute time defined by the date picker, `dashboardTime` is populated with lower threshold.

Reviewer: is lower threshold same as start time?

```js
from(bucket: "telegraf/autogen")
  |> range(start: dashboardTime)
  |> filter(fn: (r) => r._measurement == "cpu" and r._field == "usage_system")
```

{{% note %}}
To use the date picker to specify a time range in the past without "now", use `dashboardTime` as the lower limit and
[`upperDashboardTime`](#upperdashboardtime-flux) as the upper limit.
{{% /note %}}

### upperDashboardTime{id="upperdashboardtime-flux"}

> The `upperDashboardTime` template variable is defined by the upper time limit specified using the date picker.

Reviewer: upper time limit...same as stop time? Just curious if we're using consistent verbiage (if it makes sense to do so)

```

#### Database variable use cases

> Database template variables are good when visualizing multiple databases with similar or identical data structures.
> They allow you to quickly switch between visualizations for each of your databases.

Reviewer: ```suggestion
Use database template variables when visualizing multiple databases with similar or identical data structures.
Variables let you quickly switch between visualizations for each of your databases.
```

value4
```

{{% note %}}

> Since string field values [require single quotes in InfluxQL](/{{< latest "influxdb" "v1" >}}/troubleshooting/frequently-asked-questions/#when-should-i-single-quote-and-when-should-i-double-quote-in-queries),

Reviewer: ```suggestion
String field values [require single quotes in InfluxQL](/{{< latest "influxdb" "v1" >}}/troubleshooting/frequently-asked-questions/#when-should-i-single-quote-and-when-should-i-double-quote-in-queries),
```

> string values should be wrapped in single quotes.

Reviewer: ```suggestion
so wrap string values in single quotes.
```

<img src="/img/chronograf/1-6-template-vars-map-dropdown.png" style="width:100%;max-width:140px;" alt="Map variable dropdown"/>

{{% note %}}

> If values are meant to be used as string field values, wrap them in single quote ([required by InfluxQL](/{{< latest "influxdb" "v1" >}}/troubleshooting/frequently-asked-questions/#when-should-i-single-quote-and-when-should-i-double-quote-in-queries)). This only pertains to values. String keys do not matter.

Reviewer: ```suggestion
Wrap string field values in single quotes ([required by InfluxQL](/{{< latest "influxdb" "v1" >}}/troubleshooting/frequently-asked-questions/#when-should-i-single-quote-and-when-should-i-double-quote-in-queries)) (string field values only; string keys do not require quotes).
```

### Custom Meta Query
Vary part of a query with a customized meta query that pulls a specific array of values from InfluxDB.

> These variables let you pull a highly customized array of potential values and offer

Reviewer: ```suggestion
Custom meta query variables let you pull a highly customized array of potential values and offer
```

Chronograf uses URL query parameters (also known as query string parameters) to set both display options and template variables in the URL.
This makes it easy to share links to dashboards so they load in a specific state with specific template variable values selected.

> URL query parameters are appended to the end of the URL with a question mark (`?`) indicating beginning of query parameters.

Reviewer: ```suggestion
URL query parameters are appended to the end of the URL with a question mark (`?`) indicating the beginning of query parameters.
```

> Multiple query parameters can be chained together using an ampersand (`&`).

Reviewer: ```suggestion
Chain multiple query parameters together using an ampersand (`&`).
```

`variableValue`  
Value of the template variable.

{{% note %}}

> Whenever template variables are modified in the dashboard, the corresponding

Reviewer: ```suggestion
When template variables are modified in the dashboard, the corresponding
```
